From 81757cfbcb0a7a1786f094c560733cfd99b68d23 Mon Sep 17 00:00:00 2001
From: Madalin Bucur <madalin.bucur@nxp.com>
Date: Thu, 2 Nov 2017 17:33:43 +0200
Subject: [PATCH 427/767] of/device: fix compilation on non-ARM64 targets

Signed-off-by: Madalin Bucur <madalin.bucur@nxp.com>

Conflicts:
	drivers/of/device.c
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/of/device.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/of/device.c b/drivers/of/device.c
index f5f758308e1a..0c3283262f6b 100644
--- a/drivers/of/device.c
+++ b/drivers/of/device.c
@@ -14,7 +14,9 @@
 
 #include <asm/errno.h>
 #include "of_private.h"
+#ifdef CONFIG_FSL_MC_BUS
 #include "../staging/fsl-mc/include/mc.h"
+#endif
 
 /**
  * of_match_device - Tell if a struct device matches an of_device_id list
@@ -153,9 +155,11 @@ int of_dma_configure(struct device *dev, struct device_node *np, bool force_dma)
 	dev->coherent_dma_mask &= mask;
 	*dev->dma_mask &= mask;
 
+#ifdef CONFIG_FSL_MC_BUS
 	if (dev_is_fsl_mc(dev))
 		coherent = fsl_mc_is_dev_coherent(dev);
 	else
+#endif
 		coherent = of_dma_is_coherent(np);
 	dev_dbg(dev, "device is%sdma coherent\n",
 		coherent ? " " : " not ");
-- 
2.17.0

